lib/qbo_api/connection/oauth2.rb

```diff
@@ -1,5 +1,49 @@
 class QboApi
-  attr_accessor :access_token
+  attr_accessor :client_id, :client_secret, :refresh_proc
+
+  OAUTH2_BASE           = 'https://oauth.platform.intuit.com'
+  OAUTH2_URL            =  OAUTH2_BASE + '/oauth2/v1/tokens/bearer'
+
+  def access_token
+    @access_token || (refresh_access_token! if refresh_token)
+  end
+
+  def access_token=(token)
+    reset_connections
+    @access_token = token
+  end
+
+  def refresh_token
+    @refresh_token
+  end
+
+  def refresh_token=(token)
+    reset_connections if refresh_token
+    refresh_proc.call(token) if refresh_proc and refresh_token
+    @refresh_token = token
+  end
+
+  def refresh_access_token!
+    resp = oauth2_connection.post do |req|
+      req.body = {
+          grant_type: :refresh_token,
+          refresh_token: refresh_token
+      }
+    end
+    data = parse_response_body(resp)
```

Comment:
`        conn.use FaradayMiddleware::ParseJson, parser_options: { symbolize_names: true }` for the win

Reply:
Should we start another issue to discuss using that middleware across the gem? I've been wondering why we're not using that.

Reply:
Yes.  would have to make it configurable since default would be symbolize_names: false

But, I believe the reason is that unless you know you're making a json request, you can't add it, and some of the request are XML